### PR TITLE
test: swapExactAmountIn happyPath

### PR DESCRIPTION
**Pull request by 0xAustrian**
Closes BAL-32

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ src = 'src/interfaces/'
 
 [fuzz]
 runs = 1000
-max_test_rejects = 20_000_000
+max_test_rejects = 1000000
```
Comment (wei3erHase) [inline foundry.toml:29]: i liked the 1_000_000 format

File: test/unit/BPool.t.sol [hunk 2]
```diff
@@ -500,7 +505,94 @@ contract BPool_Unit_ExitPool is BasePoolTest {
...
+
+    // max - calcSpotPrice
+    vm.assume(_fuzz.tokenInBalance < type(uint256).max / _fuzz.tokenInDenorm);
+    vm.assume(_fuzz.tokenOutBalance < type(uint256).max / _fuzz.tokenOutDenorm);
+
+    // internal calculation for calcSpotPrice
+    uint _numer = bdiv(_fuzz.tokenInBalance, _fuzz.tokenInDenorm);
+    uint _denom = bdiv(_fuzz.tokenOutBalance,  _fuzz.tokenOutDenorm);
+    uint _ratio = bdiv(_numer, _denom);
+    uint _scale = bdiv(BONE, bsub(BONE, MIN_FEE));
```
Comment (wei3erHase) [inline test/unit/BPool.t.sol:571]: 256!

File: test/unit/BPool.t.sol [hunk 3]
```diff
@@ -61,10 +62,18 @@ abstract contract BasePoolTest is Test, BConst, Utils {
     bPool.set__records(_token, _record);
   }
 
+  function _setBalance(address _user, uint256 _balance) internal {
+    _writeUintAtAddressMapping(address(bPool), 0, _user, _balance);
+  }
+
```
Comment (wei3erHase) [inline test/unit/BPool.t.sol:68]: i don't think we're using this one..

Comment (0xAustrian) [reply test/unit/BPool.t.sol]: Fixed!

**Review by wei3erHase — APPROVED** [review]
wonder-full 🪄

**Pull request merged**